File: airbyte-commons/src/main/java/io/airbyte/commons/util/AirbyteStreamAware.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.util;

import io.airbyte.protocol.models.AirbyteStreamNameNamespacePair;
import java.util.Optional;

/**
 * Interface that indicates that an object exposes information used to identify an Airbyte stream.
 */
public interface AirbyteStreamAware {

  /**
   * Returns the {@link AirbyteStreamNameNamespacePair} identifying the Airbyte stream associated with
   * the object.
   *
   * @return The {@link AirbyteStreamNameNamespacePair} identifying the Airbyte stream (may be empty).
   */
  default Optional<AirbyteStreamNameNamespacePair> getAirbyteStream() {
    return Optional.empty();
  }

}

File: airbyte-cdk/java/airbyte-cdk/dependencies/bin/main/io/airbyte/commons/util/AirbyteStreamAware.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.commons.util

import io.airbyte.protocol.models.AirbyteStreamNameNamespacePair
import java.util.*

/**
 * Interface that indicates that an object exposes information used to identify an Airbyte stream.
 */
interface AirbyteStreamAware {
    /**
     * Returns the [AirbyteStreamNameNamespacePair] identifying the Airbyte stream associated with
     * the object.
     */
    val airbyteStream: Optional<AirbyteStreamNameNamespacePair>
        get() = Optional.empty()
}
